### PR TITLE
Refactors config structures to follow golang conventions

### PR DESCRIPTION
**Pull request by josejulio**
Using struct tags to define the yaml mappings

More info: https://github.com/swift-sunshine/swscore/pull/30#discussion_r168771530

### CODE REVIEW

**Comment by josejulio** [comment]
> @josejulio can you add some of these into the config_test.go ? I don't think what you did is going to be right. I think you have to use the notation "'yaml: promethsu_server_url; omitempty" or something like that.
> 
> See the yaml docs: https://godoc.org/gopkg.in/yaml.v2

the `yaml` seems to be key of the `tag` [1], but it looks like its optional for yaml. Added a commit with a test for that.

I could make it explicit if you like to.

[1] https://stackoverflow.com/a/30889373/688379

**Comment by josejulio** [comment]
It's optional to specify the `yaml` key if is the only tag, look here [1]
```go
		tag := field.Tag.Get("yaml")
		if tag == "" && strings.Index(string(field.Tag), ":") < 0 {
			tag = string(field.Tag)
		}
```

[1] https://github.com/go-yaml/yaml/blob/v2/yaml.go#L242-L245

**Comment by jmazzitelli** [comment]
OK, as long as we add tests to show the yaml read in properly, that's all I care about. Whether yaml is specified or not, doesn't matter.

**Comment by josejulio** [comment]
The test for that is here: https://github.com/swift-sunshine/swscore/pull/33/files#diff-bfeae88573cf9afa7638e19a4f7b4535R64

File: config/config_test.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ func TestMarshalUnmarshal(t *testing.T) {
 		Server: Server{
 			Address: "foo-test",
 			Port:    321,
+			StaticContentRootDirectory: "/tmp",
```
Comment (jmazzitelli) [inline config/config_test.go]: This isn't testing what we need to test. We need to test going FROM YAML to data.

So you need to test that this parses correctly:

```
server:
   static_content_root_directory: /tmp
```

Comment (josejulio) [reply config/config_test.go]: Below is marshaled and unmarshaled before the test.

Comment (jmazzitelli) [reply config/config_test.go]: yeah, but for all you know, the marshalling turned the propery name into "staticcontentrootdirectory" which then gets properly read in.

We need to test that that "key;omitempty" is actually doing what we expect - that the yaml actually is the one with the underscores.

Comment (josejulio) [reply config/config_test.go]: Just uploaded a new test, not sure if is the right way to test this.

File: config/config_test.go [hunk 2]
```diff
@@ -33,11 +33,28 @@ func TestDefaults(t *testing.T) {
...
+	}
+
+	yamlString, err := Marshal(&testConf)
+	if err != nil {
+		t.Errorf("Failed to marshal: %v", err)
+	}
+	if yamlString != "server:\n  static_content_root_directory: /tmp\n" {
+		t.Errorf("Failed to marshal - StaticContentRootDirectory to static_content_root_directory: [%v]", yamlString)
+	}
+}
```
Comment (jmazzitelli) [inline config/config_test.go]: Looks good. This tests marshalling with the underscores. Now just add an unmarshal to make sure the underscores in the yaml goes the other way. Something like:

```
conf2, err := Unmarshal(yamlString)
if err != nil {
    t.Errorf("Failed to unmarshal: %v", err)
}
if (conf2.Server.StaticContentRootDirectory != "/tmp") {
    t.Errorf("Failed to unmarshal root dir")
}
```

I think the other test might confirm this, but this makes it more explicit what the test is trying to do. I specifically want to see the yaml with underscores getting marshaled then unmarshaled.

Comment (josejulio) [reply config/config_test.go]: > I think the other test might confirm this, but this makes it more explicit what the test is trying to do. I specifically want to see the yaml with underscores getting marshaled then unmarshaled.

Thought the same. Will make it explicit in the same test.

**Pull request merged**